// @prisma/generated/type-graphql/resolvers/relations/CustomerContact/CustomerContactRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerContactRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
const Customer_1 = require("../../../models/Customer");
const CustomerContact_1 = require("../../../models/CustomerContact");
const State_1 = require("../../../models/State");
const helpers_1 = require("../../../helpers");
let CustomerContactRelationsResolver = class CustomerContactRelationsResolver {
    async customer(customerContact, ctx, info) {
        const { _count } = (0, helpers_1.transformInfoIntoPrismaArgs)(info);
        return (0, helpers_1.getPrismaFromContext)(ctx).customerContact.findUniqueOrThrow({
            where: {
                id: customerContact.id,
            },
        }).customer({
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async state(customerContact, ctx, info) {
        const { _count } = (0, helpers_1.transformInfoIntoPrismaArgs)(info);
        return (0, helpers_1.getPrismaFromContext)(ctx).customerContact.findUniqueOrThrow({
            where: {
                id: customerContact.id,
            },
        }).state({
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
};
exports.CustomerContactRelationsResolver = CustomerContactRelationsResolver;
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => Customer_1.Customer, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Info()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [CustomerContact_1.CustomerContact, Object, Object]),
    tslib_1.__metadata("design:returntype", Promise)
], CustomerContactRelationsResolver.prototype, "customer", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => State_1.State, {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Info()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [CustomerContact_1.CustomerContact, Object, Object]),
    tslib_1.__metadata("design:returntype", Promise)
], CustomerContactRelationsResolver.prototype, "state", null);
exports.CustomerContactRelationsResolver = CustomerContactRelationsResolver = tslib_1.__decorate([
    TypeGraphQL.Resolver(_of => CustomerContact_1.CustomerContact)
], CustomerContactRelationsResolver);
