// @prisma/generated/type-graphql/resolvers/outputs/StateCount.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StateCount = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
const StateCountCustomerArgs_1 = require("./args/StateCountCustomerArgs");
const StateCountCustomerContactArgs_1 = require("./args/StateCountCustomerContactArgs");
const StateCountUserArgs_1 = require("./args/StateCountUserArgs");
let StateCount = class StateCount {
    getUser(root, args) {
        return root.user;
    }
    getCustomer(root, args) {
        return root.customer;
    }
    getCustomerContact(root, args) {
        return root.customerContact;
    }
};
exports.StateCount = StateCount;
tslib_1.__decorate([
    TypeGraphQL.Field(_type => TypeGraphQL.Int, {
        name: "user",
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [StateCount, StateCountUserArgs_1.StateCountUserArgs]),
    tslib_1.__metadata("design:returntype", Number)
], StateCount.prototype, "getUser", null);
tslib_1.__decorate([
    TypeGraphQL.Field(_type => TypeGraphQL.Int, {
        name: "customer",
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [StateCount, StateCountCustomerArgs_1.StateCountCustomerArgs]),
    tslib_1.__metadata("design:returntype", Number)
], StateCount.prototype, "getCustomer", null);
tslib_1.__decorate([
    TypeGraphQL.Field(_type => TypeGraphQL.Int, {
        name: "customerContact",
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [StateCount, StateCountCustomerContactArgs_1.StateCountCustomerContactArgs]),
    tslib_1.__metadata("design:returntype", Number)
], StateCount.prototype, "getCustomerContact", null);
exports.StateCount = StateCount = tslib_1.__decorate([
    TypeGraphQL.ObjectType("StateCount", {
        simpleResolvers: true
    })
], StateCount);
