// @prisma/generated/type-graphql/enums/CustomerScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
var CustomerScalarFieldEnum;
(function (CustomerScalarFieldEnum) {
    CustomerScalarFieldEnum["id"] = "id";
    CustomerScalarFieldEnum["name"] = "name";
    CustomerScalarFieldEnum["streetAddress"] = "streetAddress";
    CustomerScalarFieldEnum["streetAddress2"] = "streetAddress2";
    CustomerScalarFieldEnum["city"] = "city";
    CustomerScalarFieldEnum["stateId"] = "stateId";
    CustomerScalarFieldEnum["zip"] = "zip";
    CustomerScalarFieldEnum["phone"] = "phone";
    CustomerScalarFieldEnum["dateCreated"] = "dateCreated";
    CustomerScalarFieldEnum["dateUpdated"] = "dateUpdated";
})(CustomerScalarFieldEnum || (exports.CustomerScalarFieldEnum = CustomerScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(CustomerScalarFieldEnum, {
    name: "CustomerScalarFieldEnum",
    description: undefined,
});
