// @prisma/generated/type-graphql/enums/CustomerContactScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerContactScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
var CustomerContactScalarFieldEnum;
(function (CustomerContactScalarFieldEnum) {
    CustomerContactScalarFieldEnum["id"] = "id";
    CustomerContactScalarFieldEnum["customerId"] = "customerId";
    CustomerContactScalarFieldEnum["firstName"] = "firstName";
    CustomerContactScalarFieldEnum["lastName"] = "lastName";
    CustomerContactScalarFieldEnum["email"] = "email";
    CustomerContactScalarFieldEnum["streetAddress"] = "streetAddress";
    CustomerContactScalarFieldEnum["streetAddress2"] = "streetAddress2";
    CustomerContactScalarFieldEnum["city"] = "city";
    CustomerContactScalarFieldEnum["stateId"] = "stateId";
    CustomerContactScalarFieldEnum["zip"] = "zip";
    CustomerContactScalarFieldEnum["phone"] = "phone";
    CustomerContactScalarFieldEnum["dateCreated"] = "dateCreated";
    CustomerContactScalarFieldEnum["dateUpdated"] = "dateUpdated";
})(CustomerContactScalarFieldEnum || (exports.CustomerContactScalarFieldEnum = CustomerContactScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(CustomerContactScalarFieldEnum, {
    name: "CustomerContactScalarFieldEnum",
    description: undefined,
});
