// @prisma/generated/type-graphql/enums/UserScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
var UserScalarFieldEnum;
(function (UserScalarFieldEnum) {
    UserScalarFieldEnum["id"] = "id";
    UserScalarFieldEnum["firstName"] = "firstName";
    UserScalarFieldEnum["lastName"] = "lastName";
    UserScalarFieldEnum["email"] = "email";
    UserScalarFieldEnum["streetAddress"] = "streetAddress";
    UserScalarFieldEnum["streetAddress2"] = "streetAddress2";
    UserScalarFieldEnum["city"] = "city";
    UserScalarFieldEnum["stateId"] = "stateId";
    UserScalarFieldEnum["zip"] = "zip";
    UserScalarFieldEnum["phone"] = "phone";
    UserScalarFieldEnum["password"] = "password";
    UserScalarFieldEnum["role"] = "role";
    UserScalarFieldEnum["dateCreated"] = "dateCreated";
})(UserScalarFieldEnum || (exports.UserScalarFieldEnum = UserScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(UserScalarFieldEnum, {
    name: "UserScalarFieldEnum",
    description: undefined,
});
